app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt

```
@@ -122,65 +117,52 @@ class SignUpViewModel(
                         val errorBody = response.errorBody()!!.string()
                         val jsonObject = JSONObject(errorBody)
                         val message = jsonObject.getString("message")
 
-                        ErrorUIState.Error(message = message)
+                        sharedViewModel.setErrorState(ErrorUIState.Error(message = message))
                     }
                 }
             } catch (e: IOException) {
-                errorUIState = ErrorUIState.Error(R.string.error_network)
+                sharedViewModel.setErrorState(ErrorUIState.Error(R.string.error_network))
             } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)